lib/runner.js

```diff
@@ -704,6 +704,9 @@ Runner.prototype.uncaught = function (err) {
     debug('uncaught undefined exception');
     err = undefinedError();
   }
+  if (!(err instanceof Error || (err && typeof err.message === 'string'))) {
+    err = new Error('the ' + type(err) + ' ' + stringify(err) + ' was thrown, throw an Error :)');
```

Comment:
Maybe create a `string2Error` function and have both `.uncaught()` and `.fail()` [[here](https://github.com/mochajs/mocha/blob/cc2bdf6c73f8e333f09fec83465bc4130ac8371a/lib/runner.js#L228)] call it? DRY.